### PR TITLE
XmlWriter respects NewLineOnAttributes when writing xmlns

### LINKED ISSUE #74840: `XmlWriterSettings.NewLineOnAttributes` is ignored on writing `xmlns:*` attributes
### Description

As described in the title.
`XmlWriterSettings.NewLineOnAttributes` is applied to normal attributes properly, but seems not to be applied to `xmlns` attributes.

This problem seems to occur with both case of `xmlns="..."` and `xmlns:nsname="..."`.

### Reproduction Steps

Code for reproduction:
```cs
using System.Xml;
using System.Xml.Linq;

var nsSvg = (XNamespace)"http://www.w3.org/2000/svg";
var nsXLink = (XNamespace)"http://www.w3.org/1999/xlink";
var doc = new XDocument(
  new XDeclaration("1.0", "utf-8", null),
  new XElement(
    nsSvg + "svg",
    new XAttribute("version", "1.1"),
    new XAttribute("width", "10"),
    new XAttribute(XNamespace.Xmlns + "xlink", nsXLink.NamespaceName),
    new XAttribute("height", "10"),
    new XElement(nsSvg + "g")
  )
);

var settings = new XmlWriterSettings() {
  CloseOutput = false,
  Indent = true,
  IndentChars = " ",
  NewLineChars = "\n",
  NewLineOnAttributes = true
};

using (var writer = XmlWriter.Create(Console.Out, settings)) {
  doc.Save(writer);
}

Console.WriteLine();
```

Note: The same problem occurs  with `XDocument` created from `XDocument.Load`.

```cs
var reader = new StringReader(@"<?xml version=""1.0"" encoding=""utf-8""?>
<svg
 version=""1.1""
 xmlns=""http://www.w3.org/2000/svg""
 width=""10""
 xmlns:xlink=""http://www.w3.org/1999/xlink""
 height=""10""
>
 <g />
</svg>
");
var doc = XDocument.Load(reader);
```

### Expected behavior

```xml
<?xml version="1.0" encoding="utf-8"?>
<svg
 version="1.1"
 width="10"
 xmlns:xlink="http://www.w3.org/1999/xlink"
 height="10"
 xmlns="http://www.w3.org/2000/svg">
 <g />
</svg>
```

### Actual behavior

```xml
<?xml version="1.0" encoding="utf-8"?>
<svg
 version="1.1"
 width="10" xmlns:xlink="http://www.w3.org/1999/xlink"
 height="10" xmlns="http://www.w3.org/2000/svg">
 <g />
</svg>
```

### Regression?

_No response_

### Known Workarounds

_No response_

### Configuration

```
$ dotnet --info
.NET SDK (reflecting any global.json):
 Version:   6.0.400
 Commit:    7771abd614

Runtime Environment:
 OS Name:     ubuntu
 OS Version:  20.04
 OS Platform: Linux
 RID:         ubuntu.20.04-x64
 Base Path:   /usr/share/dotnet/sdk/6.0.400/

global.json file:
  Not found

Host:
  Version:      6.0.8
  Architecture: x64
  Commit:       55fb7ef977

*snip*
```

### Other information

_No response_

### LINKED ISSUE #80306: `XmlWellFormedWriter` ignores `XmlWriterSettings.NewLineOnAttributes` for attributes specifically named `xmlns`
### Description

When writing an `XDocument`  to a stream via `XmlWriter`, even if specifically requested, attributes specifically named `xmlns` will not write on their own line. Other attributes don't suffer this issue.

### Reproduction Steps

See this .NET Fiddle: https://dotnetfiddle.net/PCyVdc (`.NET 7`)
(Similar fiddle for `.NET Framework 4.7.2` [here](https://dotnetfiddle.net/eN2e6p).)

<details>
<summary>Code from fiddle reproduced here.</summary>

```csharp
using System;
using System.Xml;
using System.Xml.Linq;
using System.IO;
					
public class Program
{
	public static void Main()
	{
		XmlWriterSettings writerSettings = new() {
			// enable newlines on attributes
			NewLineOnAttributes = true,
			
			// value of NewLineHandling doesn't chnage behavior
			//NewLineHandling = NewLineHandling.None,
			//NewLineHandling = NewLineHandling.Entitize,
			NewLineHandling = NewLineHandling.Replace,
			
			// value of NewLineChars doesn't change behavior
			NewLineChars = "\n",
			
			// if Indent is false, no newlines will be written at all,
			// regardless of NewLineHandling; which seems improper 🤷🏻‍♀️
			Indent = true,
		};
		
		string[] tests = new[] {
			// this test will properly writes each attribute on their own lines
			"""<?xml version="1.0" encoding="utf-8"?><data attrib1="1" attrib2="2" attrib3="3" attrib4="4"><child /></data>""",
			// xmlns is written next to the element name, but other attributes are correctly written on their own line
			"""<?xml version="1.0" encoding="utf-8"?><data xmlns="1" attrib2="2" attrib3="3" attrib4="4"><child /></data>""",
			// no newline after attrib3, xmlns is written on the same line
			"""<?xml version="1.0" encoding="utf-8"?><data attrib1="1" attrib2="2" attrib3="3" xmlns="4"><child /></data>""",
			// no newline after attrib2, xmlns is written on the same line, but attrib4 is correctly written on it's own line
			"""<?xml version="1.0" encoding="utf-8"?><data attrib1="1" attrib2="2" xmlns="3" attrib4="4"><child /></data>""",
			// the same as above, just on an empty element
			"""<?xml version="1.0" encoding="utf-8"?><data attrib1="1" attrib2="2" xmlns="3" attrib4="4" />""",
		};
		
		foreach (string test in tests)
		{
			FormatAndPrint(test, writerSettings);
		}
	}
	
	public static void FormatAndPrint(string s, XmlWriterSettings writerSettings)
	{
		// load the xml string, and write it to memory
		XDocument xd = XDocument.Parse(s);
		using MemoryStream stream = new();
		using XmlWriter writer = XmlWriter.Create(stream, writerSettings);
		xd.WriteTo(writer);
		writer.Flush();

		// read the written xml back out out of memory, and write it to console
		stream.Position = 0;
		using StreamReader reader = new StreamReader(stream, writerSettings.Encoding);
		string xml = reader.ReadToEnd();
		Console.WriteLine("---");
		Console.WriteLine(xml);
		Console.WriteLine();
	}
}
```

</details>

### Expected behavior

All written attributes should be on their own lines when `XmlWriterSettings.NewLineOnAttributes` is true.

### Actual behavior

Attributes specifically named `xmlns` are always written on the same line the writer is currently on, regardless of `XmlWriterSettings` member values.

### Regression?

Unlikely. I was able to reproduce it against the following runtimes: `.NET 7`, `.NET 6`, `.NET 5`, `.NET Core 3.1`, `.NET Framework 4.7.2`

### Known Workarounds

Manually write the XML, or post-process the writer's output and manually insert a newline & indentation.

### Configuration

Example code is running against `.NET 7`.

OS/Arch unlikely to change outcome.

### Other information

None yet.

### PR DESCRIPTION
**Pull request by devsko**
Fixes #80306
Fixes #74840

I updated the tt files but did not use them because they are too far out of sync (tried to fix that also but gave up)

### CODE REVIEW

File: src/libraries/System.Private.Xml/src/System/Xml/Core/XmlUtf8RawTextWriterAsync.cs [hunk 1]
```diff
@@ -1841,6 +1846,19 @@ protected internal override async Task WriteStartAttributeAsync(string? prefix,
...
+        internal override async Task WriteStartNamespaceDeclarationAsync(string prefix)
+        {
+            CheckAsyncCall();
+            // Add indentation
+            if (_newLineOnAttributes)
+            {
+                await WriteIndentAsync().ConfigureAwait(false);
+            }
+
+            await base.WriteStartNamespaceDeclarationAsync(prefix).ConfigureAwait(false);
```
Comment (krwq) [inline src/libraries/System.Private.Xml/src/System/Xml/Core/XmlUtf8RawTextWriterAsync.cs:1859]: just curious, why not move this logic to base?

**Comment by krwq** [comment]
Thanks @devsko, overall looks good, just one question and outerloop tests need to run since XML has plenty of them

File: src/libraries/System.Private.Xml/src/System/Xml/Core/XmlUtf8RawTextWriterAsync.cs (hunk 1 above)
Comment (devsko) [reply src/libraries/System.Private.Xml/src/System/Xml/Core/XmlUtf8RawTextWriterAsync.cs:1859]: Well, I gues you know more about the design decisions of this stuff, but all the indentation is only done in these `XxxRawTextWriterIndent` classes. I assume to save some bytes when no indentation is needed.

Comment (krwq) [reply src/libraries/System.Private.Xml/src/System/Xml/Core/XmlUtf8RawTextWriterAsync.cs:1859]: Design was made long time before I joined the team 😄

**Review by krwq — APPROVED** [review]
failures seem to be unrelated

**Pull request merged**